src/docfx/build/toc/parser/TableOfContentsParser.cs

```diff
@@ -276,10 +284,17 @@ SourceInfo<string> GetTopicHref(TableOfContentsItem tocInputModel)
                 // process uid first
                 if (!string.IsNullOrEmpty(uid))
                 {
-                    var (uidHref, uidDisplayName, uidFile) = resolveXref.Invoke(rootPath, uid);
-                    if (!string.IsNullOrEmpty(uidHref))
+                    var (uidError, uidLink, display, xrefSpec) = context.DependencyResolver.ResolveXref(uid, filePath, rootPath);
```

Comment:
extract a method for `reoslveXref` and `resolveHref` and such on? `LoadTocModel` is getting quite long

Reply:
most of them are inner methods, do you mean you want to split them outof LoadTocModel and make them private static method?

---
In reply to: [296099852](https://github.com/dotnet/docfx/pull/4769#discussion_r296099852) [](ancestors = 296099852)